gridstatus/spp.py

```diff
@@ -782,6 +786,60 @@ def get_interconnection_queue(self, verbose=False):
 
         return queue
 
+    @support_date_range("DAY_START")
+    def get_lmp_real_time_5_min(self, date, end=None, verbose=False):
+        """Get LMP data for the Real-Time 5 Minute Market
+
+        Args:
+            date: date to get data for
+            end: end date
+            verbose: print url
+        """
+        year = date.strftime("%Y")
+        month = date.strftime("%m")
+        day = date.strftime("%d")
+
+        day_url = f"https://portal.spp.org/file-browser-api/download/rtbm-lmp-by-bus?path=%2F{year}%2F{month}%2FBy_Day%2FRTBM-LMP-DAILY-BUS-{year}{month}{day}.csv"
+
+        # First, check if there are daily files for the date. If not, go to
+        # 5-minute files.
```

Comment:
@kmax12 I haven't finished this yet, but what do you think of the approach of first looking for daily files, then switching to 5-min interval files if the daily file isn't available? This cuts down the number of requests by a factor of 288 for historical dates.

There are also monthly rollup files but those are fairly large so I stuck with the daily files.

Reply:
This approach proved too complicated. Now this only uses the 5-minute files.